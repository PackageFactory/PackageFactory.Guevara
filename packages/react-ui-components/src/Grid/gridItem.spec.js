import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import GridItem from './gridItem.js';

describe('<GridItem/>', () => {
    let props;

    beforeEach(() => {
        props = {
            theme: {},
            width: 'third',
            children: 'Foo children'
        };
    });

    it('should render correctly.', () => {
        const wrapper = shallow(<GridItem {...props}/>);

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should allow the propagation of "className" with the "className" prop.', () => {
        const wrapper = shallow(<GridItem {...props} className="fooClassName"/>);

        expect(wrapper.prop('className')).toContain('fooClassName');
    });

    it('should allow the propagation of additional props to the wrapper.', () => {
        const wrapper = shallow(<GridItem {...props} foo="bar"/>);

        expect(wrapper.prop('foo')).toBe('bar');
    });

    it('should render a inline style matching the passed "width" prop.', () => {
        const wrapper = shallow(<GridItem {...props} width="half"/>);

        expect(wrapper.html()).toContain('style="width:50%"');
    });
});
