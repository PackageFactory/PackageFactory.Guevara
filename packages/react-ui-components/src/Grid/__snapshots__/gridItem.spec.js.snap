// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridItem/> should render correctly. 1`] = `
<div
  className=""
  style={
    Object {
      "width": "33.33%",
    }
  }
>
  Foo children
</div>
`;
